libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -770,6 +770,16 @@ class MapboxNavigation(
         return arrivalProgressObserver.navigateNextRouteLeg()
     }
 
+    /**
+     * Follows a new leg of the already loaded directions.
+     *
+     * @param legIndex new leg index
+     * @return true if no errors occurred, false otherwise
+     */
+    fun updateLegIndex(legIndex: Int): Boolean {
+        return tripSession.updateLegIndex(legIndex)
+    }
```

Comment:
https://github.com/mapbox/mapbox-navigation-android/pull/4739#issuecomment-904128945

Add a function to `arrivalProgressObserver` and the `onNextRouteLegStart` callback will be handled for you

```suggestion
    fun updateLegIndex(legIndex: Int): Boolean {
        return arrivalProgressObserver.updateLegIndex(legIndex: Int)
    }
```

Also, could update `updateLegIndex` to `navigateLegIndex` to make it fit in with `navigateNextRouteLeg`

Reply:
``` kotlin
interface ArrivalObserver {
    /**
     * Called when [MapboxNavigation.navigateNextRouteLeg] has been called and returns true,
     * this observer will be notified of the next route leg start.
     */
    fun onNextRouteLegStart(routeLegProgress: RouteLegProgress)
```

This comment won't really make sense after this too. Will you update the comment to
``` kotlin
     * Called when [MapboxNavigation.navigateNextRouteLeg] or [MapboxNavigation. updateLegIndex]
     * has been called and returns true, this observer will be notified of the next route leg start.

Reply:
Actually, I don't think we should add this updateLegIndex function. The request, is to start the navigation trip at an index https://github.com/mapbox/mapbox-navigation-android/issues/4681

So I think we should introduce a way to start the navigation trip at an index
https://github.com/mapbox/mapbox-navigation-android/pull/4739#issuecomment-904837527